src/arb/arb.jl

```diff
@@ -85,19 +85,84 @@ characteristic(::ArbField) = 0
 #
 ################################################################################
 
-function Float64(x::arb)
-   GC.@preserve x begin
-      t = ccall((:arb_mid_ptr, libarb), Ptr{arf_struct}, (Ref{arb}, ), x)
-      # 4 == round to nearest
-      d = ccall((:arf_get_d, libarb), Float64, (Ptr{arf_struct}, Int), t, 4)
-   end
-   return d
+@doc Markdown.doc"""
+    Float64(x::arb, round::RoundingMode=RoundNearest)
+
+Converts $x$ to a `Float64`, rounded in the direction specified by $round$.
+For `RoundNearest` the return value approximates the midpoint of $x$. For
+`RoundDown` or `RoundUp` the return value is a lower bound or upper bound for
+all values in $x$.
+"""
+function Float64(x::arb, round::RoundingMode=RoundNearest)
+  t = _arb_get_arf(x, round)
+  return _arf_get_d(t, round)
+end
+
+@doc Markdown.doc"""
+    BigFloat(x::arb, round::RoundingMode=RoundNearest)
+
+Converts $x$ to a `BigFloat` of the currently used precision, rounded in the
+direction specified by $round$. For `RoundNearest` the return value
+approximates the midpoint of $x$. For `RoundDown` or `RoundUp` the return
+value is a lower bound or upper bound for all values in $x$.
+"""
+function BigFloat(x::arb, round::RoundingMode=RoundNearest)
+  t = _arb_get_arf(x, round)
+  return _arf_get_mpfr(t, round)
+end
+
+function _arb_get_arf(x::arb, ::RoundingMode{:Nearest})
+  GC.@preserve x begin
```

Comment:
Since the data to which `t` points is owned by `x`, this `GC.@preserve x` is not needed here and is needed in the caller's code.